.github/workflows/dluhc-build-and-publish.yml

```diff
@@ -0,0 +1,190 @@
+    name: lint-and-test-${{matrix.app}}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+
+      - name: Configure yarn caching
+        uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Build dependencies
+        run: yarn build:dependencies
+
+      - name: Lint
+        run: yarn ${{matrix.app}} lint
+
+      - name: Test
+        run: yarn ${{matrix.app}} test-cov
+
+      - name: Upload test results artifacts
+        uses: actions/upload-artifact@v2
+        if: ${{ success() || failure() }}
+        with:
+          name: test-results-${{matrix.app}}
+          path: ${{matrix.app}}/test-results
+          retention-days: 14
+
+      - name: Upload test coverage artifacts
+        uses: actions/upload-artifact@v2
+        if: ${{ success() || failure() }}
+        with:
+          name: test-coverage-${{matrix.app}}
+          path: ${{matrix.app}}/test-coverage
+          retention-days: 14
+
+  setup-env-vars:
+    needs: lint-and-test
+    runs-on: ubuntu-latest
+    outputs:
+      BRANCH_NAME: ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+    steps:
+      - id: set-vars
+        run: |
+          echo "::set-output name=LOWER_CASE_BRANCH::$(
+            echo ${GITHUB_REF#refs/heads/} |
+            tr "[:upper:]" "[:lower:]"
+          )"
+      - name: echo-vars
+        id: echo-vars
+        run: |
+          echo ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+          echo $VERSION
+
+  build-and-push-images:
+    needs: [lint-and-test, setup-env-vars]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        app: [designer, runner]
+    name: build-and-push-${{matrix.app}}-image
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+      - name: Docker compose pull
+        run: docker-compose pull
+      - name: Docker layer caching DAYOFYEAR key
+        # DAYOFYEAR key is used to rotate docker-layer-caching daily to avoid the layers dangle issues
+        # see https://github.com/satackey/action-docker-layer-caching/issues/55
+        # if you change this, please remember to apply the same changes to branch--lint-unit-and-smoke-test.yml
+        # so main branch cache can be shared with branches
+        run: echo "DAYOFYEAR=$(date +%j)" >> $GITHUB_ENV
```

Comment:
Is this intended to be used alongside some sort of cache action? Not sure it would do anything on its own?

Reply:
It was in the original workflow file from upstream but I couldn't see it doing anything either - the plugin that issue link relates to isn't used here. Will remove.